test/Interop/Cxx/templates/Inputs/ForwardDeclaredSpecialization.h

```diff
@@ -0,0 +1,19 @@
+#ifndef FORWARD_DECLARED_SPECIALIZATION_H
+#define FORWARD_DECLARED_SPECIALIZATION_H
+
+template <typename T> 
+struct MyTemplate { 
+  T value; 
+};
+
+template <> 
+struct MyTemplate<int>;
+typedef MyTemplate<int> MyIntTemplate;
```

Comment:
The problem here really is that the user declares the specialization `template <> struct MyTemplate<int>` without defining it. So there shouldn't be any issue if they just wrote `typedef MyTemplate<int> MyIntTemplate` without the specialization, right?

If that's correct, can you please add a test case with that scenario to emphasize this point? e.g.,

```cpp
// NOTE: no explicit specialization, i.e., template<> struct MyTemplate<bool>
typedef MyTemplate<bool> MyCompleteBoolTemplate;
```